NSURL.m

```diff
@@ -351,7 +363,41 @@ static CFTypeRef CFURLCreatePropertyForKey(CFURLRef url, CFStringRef key, CFErro
     else if (CFEqual(key, kCFURLTypeIdentifierKey))
     {
         // Key for the resource’s uniform type identifier (UTI), returned as a CFString object.
+        static dispatch_once_t pred;
+        static void* LaunchServicesHandle;
+        static CFErrorRef* LaunchServicesHandleError;
+
+        dispatch_once(&pred, ^{
+            LaunchServicesHandle = dlopen("/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/LaunchServices", RTLD_LAZY | RTLD_LOCAL);
+            if(LaunchServicesHandle == NULL)
+                dlError(LaunchServicesHandleError);
```

Comment:
I believe you're currently passing an uninitialized pointer into `dlError`? But it seems like what you want instead is:
```suggestion
        static CFErrorRef LaunchServicesHandleError;

        dispatch_once(&pred, ^{
            LaunchServicesHandle = dlopen("/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/LaunchServices", RTLD_LAZY | RTLD_LOCAL);
            if(LaunchServicesHandle == NULL)
                dlError(&LaunchServicesHandleError);
```